.azure-pipelines/changes.yml

```diff
@@ -17,9 +18,21 @@ jobs:
     job_name: Changed
     display: Linux
     validate: true
+    pip_cache_config:
+      key: 'pip | $(Agent.OS) | datadog_checks_base/datadog_checks/base/data/agent_requirements.in'
+      restoreKeys: |
+        pip | $(Agent.OS)
+        pip
+      path: $(PIP_CACHE_DIR)
 
 - template: './templates/test-single-windows.yml'
   parameters:
     job_name: Changed
     check: '--changed datadog_checks_base datadog_checks_dev active_directory aspdotnet disk dotnetclr exchange_server iis pdh_check sqlserver win32_event_log windows_service wmi_check'
     display: Windows
+    pip_cache_config:
+      key: 'pip | $(Agent.OS) | datadog_checks_base/datadog_checks/base/data/agent_requirements.in'
+      restoreKeys: |
+        pip | $(Agent.OS)
+        pip
```

Comment:
Can this cause issues? My understanding of Azure docs is that if there is no hit on the previous key, the CI will try to restore from the latest key that starts with the prefixes.

I understand the line before (if the agent_requirements.in file changes, it's a good idea to reuse the part of the cache that we have). But I'm not sure of the effects of sharing a cache between Linux and Windows

Reply:
Interesting point! Indeed I'm not sure of the effects of sharing cache items between Linux/Windows either. We can probably leave the `pip` restore key out for now.